### DISCUSSION TITLE
Anyone on macOS Sonoma Beta right now?

### DISCUSSION THREAD

**Discussion by vincentneo** [vincentneo/LosslessSwitcher#96] [Q&A] [answered]
If you are / know someone that is using it,
Please help me test: 
1. Does LosslessSwitcher still work there?
2. Is LosslessSwitcher still needed (does Music app natively support sample rate switching?)

Thank you!

**Comment by MasterRahool** [comment]
I havent installed it yet but I'd put money on (2) being a negative, havent heard it mentioned from Music beta users.

**Reply by vincentneo** [reply]
Same thoughts too

**Comment by PivotMasterNM** [comment, marked as answer]
On the Sonoma Beta since day 1!

MacBook Pro 16 M1Max, current Public Beta Build 3 14.0 ( 23A5312d ).

1.) Seems to work and behave just like in Ventura, i.e. it works great when using the integrated DAC Headphone Jack, and behaves more stuttery depending on what external DAC is connected. For me it seems pretty much identical so far.

2.) Couldn't observe any switching, seems to behave like in Ventura too.

**Reply by vincentneo** [reply]
Thanks for sharing this!

I was initially optimistic that Sonoma contains a Music app that does the switching by itself, since I haven't heard about anything regarding users of macOS Sonoma betas until now. 

Guess LosslessSwitcher lives on for another year!

**Reply by PivotMasterNM** [reply]
I suspect that this change will probably never be implemented in macOS.

On iOS, apps have full foreground focus and there is no simultaneous playback of any audio allowed besides the apple music app. Switching Sample Rates and Bit Depths is therefore much easier and has no potential of interfering with other apps, and can me done much faster and simpler, which makes it basically unnoticeable to the user. Plus: there are iOS frameworks for audio output and playback, macOS does not require that. WIth iOS, there is full control.

On macOS, when often switching Sample Rates (and/OR disconnecting an external DAC) regularly can really mess things up for many apps. YouTube Playback in safari might crash, might have mismatched sample rates and not get resampled, so there is pitch shifting and/or glitchin when playing back. Discord can make your sound robotic. Zoom might just not connect you. Some programs just straight up crash.

Source: I am an audio engineer, sample rate switching has haunted my life (audio industry uses 44.1, movie/TV likes to use 48 ). Have experiences all of the above on macOS, on windows its even worse (thanks DirectWave).

Sample rate converters have gotten so good, that the normal consumer doesnt care and wont be able to hear the difference. None of us Audiophiles would probably be able to hear the difference when ABX testing - but I do believe if it's possible, why not have the full quality available.

Hope apple implements it someday! Just don't know why they would or how...

**Reply by vincentneo** [reply]
> On iOS, apps have full foreground focus and there is no simultaneous playback of any audio allowed besides the apple music app.

Meanwhile, at least with tvOS 16 and below (not sure about 17), which I assume is much similar to iOS, there is no Hi-Res Lossless mode for Apple Music at all (option only exists for up to 48kHz). I guess it has something to do with DVD/movie preference of 48kHz?

> On macOS, when often switching Sample Rates (and/OR disconnecting an external DAC) regularly can really mess things up for many apps.

And, yes... I do see that sometimes 😰

> None of us Audiophiles would probably be able to hear the difference when ABX testing - but I do believe if it's possible, why not have the full quality available.

Agreed!
